WIP: Navigation

SyrianSpock opened the pull request:
Achieved:
- Navigation in simulation
- Navigation on robot but with navigation stack running on PC

To do:
- Try running navigation stack directly on robot
- Test static obstacle avoidance
- Tune planners and controllers

SyrianSpock committed:
Start from URDF tutorial (R2D2 model)

SyrianSpock committed:
Strap robot model down to differential wheelbase

SyrianSpock committed:
Add 1D lidar to robot simulation

SyrianSpock committed:
Checkin Rviz config for simulation

SyrianSpock committed:
Publish robot TF using URDF

SyrianSpock committed:
Setup visualisation config for monitoring

SyrianSpock committed:
Write a basic config for navigation on simulation

Use a dummy test map and use default parameters for the costmap and planners

SyrianSpock committed:
Fix oscillation of the robot in simulation

Also scale up the base link

SyrianSpock committed:
Add caster links to balance the wheelbase

SyrianSpock committed:
Add launch file for real world navigation

SyrianSpock committed:
Add missing Cmakelists

SyrianSpock committed:
Separate robot model from simulation

Create a dedicated package for the URDF description

SyrianSpock committed:
Avoid publishing static joints position

SyrianSpock committed:
Fix navigation stack setup

SyrianSpock committed:
Write a simple script to simulate a perfect motor node

SyrianSpock committed:
Add a launch file that runs the robot with simulated motors

SyrianSpock committed:
Fix build & Document new dependencies

SyrianSpock committed:
Provide default parameters to controller & odometry nodes

SyrianSpock committed:
Tone down local planner parameters for testing/debugging

antoinealb commented:
Any reason that the navigation stack cannot run on the BBB for now ?

Reading the code now.

SyrianSpock commented:
I just didn't have the time.

antoinealb commented on goldorak_simulation/config/control.yaml:
```diff
@@ -0,0 +1,41 @@
+
+  # Velocity and acceleration limits for the robot
+  linear:
+    x:
```
What is the difference between this and the limits defined in `goldorak_navigation/param/base_local_planner_params.yaml` ? Is that one purely the simulation limit ?

SyrianSpock replied:
These are implemented as clamps on the simulated controller side.
It's not used by the "real controller" on the robot, although we could benefit from implementing clamps.

The others in `goldorak_navigation/param/base_local_planner_params.yaml` are limitations imposed on the planner that model the dynamics of the system more accurately: static friction is modeled with the minimum velocity applicable.

SyrianSpock replied:
Actually the "real controller" is not even a controller, it just converts velocity commands from the planner into wheel setpoints, so it's basically open-loop. Maybe we should implement a PID to follow the commands with better performance.

SyrianSpock committed:
Move full stack simulation launch file to simulation package

SyrianSpock committed:
Refactor deployment/navigation launch files into one bringup launch

SyrianSpock committed:
Rename joint state publisher node more explicitly

SyrianSpock committed:
Signal nodes readiness in ROS logger

SyrianSpock committed:
Clean up bringup launch file

SyrianSpock committed:
Update Rviz robot monitoring configuration to display navigation data

SyrianSpock committed:
Explore some configurations for navigation

SyrianSpock committed:
Try out turtlebot navigation setup

Works but robot oscillates sometimes

SyrianSpock committed:
Make motor simulation script more accurate

Integrate velocity correctly to compute position and encoder value of wheel

SyrianSpock committed:
Update Rviz config to view table from top

SyrianSpock committed:
Try tuning the navigation parameters, looks better

SyrianSpock committed:
Update motor simulator to use stamped encoder messages

Conflicts:
	uavcan_core/scripts/motor_board_sim.py

SyrianSpock committed:
Checkin Eurobot2016 draft map

SyrianSpock commented:
I think we should run this on hardware ASAP and merge.

SyrianSpock committed:
Update readme

antoinealb commented:
agree. wanna try on wednesday ?

SyrianSpock committed:
Fix local planner static map

Fixes weird behavior when the robot is asked to go out of table
Robot respects walls now

SyrianSpock committed:
Remove deprecated navfn planner parameters

SyrianSpock committed:
Tweaking navigation parameters

SyrianSpock committed:
Clamp simulated motor velocity

SyrianSpock committed:
Checkin missing map with static obstacles

SyrianSpock committed:
Implement a basic driver to publish 1D laser range

Using IFM 1D rangefinder wired on an ADC of the board

SyrianSpock committed:
Split up launch files to cleanup

SyrianSpock committed:
Tune navigation config to avoid running into obstacle corners

This was a problem before as the local planner would derive a lot from the global path and run into corners of the static obstacles

SyrianSpock committed:
Clean up launch files

Especially simulation launch files
Introduce cmd_vel topic as argument to launch navigation stack

SyrianSpock committed:
Document how to run simulation with Gazebo

antoinealb committed:
Start working on strategy
